htslib/kstring.h

```diff
@@ -132,16 +148,50 @@ static inline int ks_resize(kstring_t *s, size_t size)
 	return 0;
 }
 
+/// Increase kstring capacity by a given number of bytes
+static inline int ks_expand(kstring_t *s, size_t expansion)
+{
+    size_t new_size = s->l + expansion;
+
+    if (new_size < s->l) // Overflow check
+        return -1;
+    return ks_resize(s, new_size);
+}
+
+/// Returns the kstring buffer
 static inline char *ks_str(kstring_t *s)
 {
 	return s->s;
 }
 
+/// Returns the kstring buffer, or an empty string if l == 0
+/**
+ * Unlike ks_str(), this function will never return NULL.  If the kstring is
+ * empty it will return a read-only empty string.  As the returned value
+ * may be read-only, the caller should not attempt to modify it.
+ */
+static inline const char *ks_c_str(kstring_t *s)
+{
+    return s->l && s->s ? s->s : "";
+}
+
 static inline size_t ks_len(kstring_t *s)
 {
 	return s->l;
 }
 
+/// Reset kstring length to zero
+/**
+   @return The kstring itself
+
+   Example use: kputsn(string, len, ks_reset(s))
```

Comment:
s/reset/clear/ here too (and there is an excess space in `kstring_t * ks_clear` below).